src/control/system/membership.go

```diff
@@ -448,63 +448,78 @@ func (m *Membership) CheckHosts(hosts string, ctlPort int) (*RankSet, *hostlist.
 }
 
 // OnEvent handles events on channel and updates member states accordingly.
-func (m *Membership) OnEvent(_ context.Context, evt events.Event) {
-	if common.InterfaceIsNil(evt) {
+func (m *Membership) OnEvent(_ context.Context, evi events.Event) {
+	if common.InterfaceIsNil(evi) {
 		m.log.Error("nil event")
 		return
 	}
+	// FIXME: Get rid of events.Event and switch everything to use the events.RASEvent
```

Comment:
moving encode decode into RASEvent is fine but we lose the ability to process other types of events, do we want to remove Event Interface and preclude use of any event type that is not RAS? are we sure the only event type we will ever want to process is RAS?